lib/cli/transaction.rs

```diff
@@ -325,3 +332,49 @@ pub fn make_transaction_builder(
         }
     }
 }
+
+/// Retrieves a `SecretKey` based on the provided secret key string and configuration options.
+///
+/// # Arguments
+///
+/// * `secret_key` - A string representing the secret key. If empty, a `None` option is returned.
```

Comment:
It could also return an error in case `allow_unsigned_deploy` is `false`.

Reply:
I believe you meant to get clearer comment ? Please check this commit 

https://github.com/casper-ecosystem/casper-client-rs/pull/153/commits/58a301319d38a200832944d1b8be529f427fb08b

thanks

Reply:
yes, zactly